Jason/cse 429 valueerror invalid header value (#1049)
returntocorp/semgrep-docs: docs/kb/semgrep-code/InvalidHeaderValue.md

---

> description: What is this Ivalid header value error

Reviewer: ```suggestion
description: Troubleshoot "invalid header value" errors in GitHub and Gitlab
```

tags:
  - Scanning
  - Semgrep Code
  - Secrets
---

# Troubleshooting `ValueError: Invalid header value`

> When trying to scan, you may run into the following error:

Reviewer: ```suggestion
When scanning with Semgrep, you may run into the following error:
```

                       ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 415, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 244, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.11/http/client.py", line 1283, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.11/http/client.py", line 1324, in _send_request
    self.putheader(hdr, value)
  File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 224, in putheader
    _HTTPConnection.putheader(self, header, *values)
  File "/usr/local/lib/python3.11/http/client.py", line 1261, in putheader
    raise ValueError('Invalid header value %r' % (values[i],))
ValueError: Invalid header value b'Bearer *******************************************************' 
```

Root cause of this error would be a `\n` within the secret value or a possible misconfiguration from the copy/pasting of your `SEMGREP_APP_TOKEN`.

> ## Example fix on Github

Reviewer: ```suggestion
## Example fix on GitHub
```

> To fix on Github, go to either your organization's/repository's `Secrets and variables` section within the settings and update the value of the `SEMGREP_APP_TOKEN` to ensure it does not have an extraneous `\n`.

Reviewer: ```suggestion
To fix on Github:

1. Go to either your organization's or repository's **Settings** > **Secrets and variables** 
2. Update the value of the `SEMGREP_APP_TOKEN` to ensure it does not have an extraneous `\n`.
```

![Secrets and variables section in Settings](/static/img/kb/github-secrets.png)
![Updating Github Secret](/static/img/kb/github-update-value.png)

> ## Example fix on Gitlab

Reviewer: ```suggestion
## Example fix on GitLab
```

> To fix on Github go to your repository's `CI/CD` Settings and update the `SEMGREP_APP_TOKEN` variable directly to remove the extraneous `\n`.

Reviewer: ```suggestion
To fix on GitLab go to your repository's `CI/CD` Settings and update the `SEMGREP_APP_TOKEN` variable directly to remove the extraneous `\n`.
```
